src/code-components/RichTextArea/Editor.tsx

```diff
@@ -48,6 +59,35 @@ const Editor = forwardRef<Quill | null, EditorProps>(
     const onFocusRef = useRef(onFocus);
     const onKeyDownRef = useRef(onKeyDown);
     const onKeyUpRef = useRef(onKeyUp);
+    const onErrorRef = useRef(onError);
+    const onLoadingRef = useRef(onLoading);
+    const onDropRef = useRef(onDrop);
+
+    async function imageDropHandler(
+      type: string,
+      imageData: QuillImageData & { name?: string },
+    ) {
+      const handlerFunction = onDropRef.current;
+
+      if (!handlerFunction) return;
+
+      try {
+        onLoadingRef.current?.(true);
+        const blob = await getBlob(imageData);
+        if (!blob) throw new Error();
+
+        const name = imageData?.name ?? "unknown.jpg";
```

Comment:
If an image is a different format, won't that random name cause the image later on to be rendered incorrectly?